packages/engine/Source/DataSources/CallbackProperty.js

```diff
@@ -49,11 +50,14 @@ Object.defineProperties(CallbackProperty.prototype, {
 /**
  * Gets the value of the property.
  *
- * @param {JulianDate} time The time for which to retrieve the value.
+ * @param {JulianDate} [time=now] The time for which to retrieve the value. If omitted, now is used.
  * @param {object} [result] The object to store the value into, if omitted, a new instance is created and returned.
  * @returns {object} The modified result parameter or a new instance if the result parameter was not supplied or is unsupported.
  */
 CallbackProperty.prototype.getValue = function (time, result) {
+  if (!defined(time)) {
+    time = JulianDate.now();
```

Comment:
This will allocate a new object each time `getValue` is called without a `time` parameter. To avoid a performance hit, we should use a [scratch variable](https://github.com/CesiumGS/cesium/blob/main/Documentation/Contributors/CodingGuide/README.md#result-parameters-and-scratch-variables).

Reply:
Close! The `JulianDate.now` function is designed to take a `result` parameter. Instead of creating a new JulianDate object, it'll be copied into the existing scratch variable:

```js
const timeScratch = new JulianDate();

/**
 * Gets the value of the property.
 *
 * @param {JulianDate} [time=JulianDate.now()] The time for which to retrieve the value. If omitted, the current system time is used.
 * @param {object} [result] The object to store the value into, if omitted, a new instance is created and returned.
 * @returns {object} The modified result parameter or a new instance if the result parameter was not supplied or is unsupported.
 */
CallbackProperty.prototype.getValue = function (time, result) {
    if (!defined(time)) {
    time = JulianDate.now(timeScratch);
  }

  return this._callback(time);
};
```

Does that make sense?

Reply:
Ohh, right! This does make more sense! 

I adapted the code. Thanks for your patience!